### ISSUE TITLE
Parameter computation order needs more clarification

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/robotframework-testsuitesmanagement#97]
Several ways are available to influence the value of configuration parameters: parameter configuration files,
local configuration files and the command line. But the Robot Framework core provides another mechanism to import parameters: the variable file, e.g. with:

`--variablefile "./variables/testvariables.py"`

It has to be clarified what shall be the computation order in case of `--variablefile` is used together with

```
--variable teststring:"command line test string"
--variable config_file:"./config/exercise-07_config_variant2.json"
--variable local_config:"./localconfig/exercise-07_localconfig_bench1.json"
```

I gave it a try. Outcome:

`--variable <single variable>:...` overrules `--variablefile`

But `--variablefile` overrules `config_file` and `local_config`.

This means: The Robot Framework internal `--variablefile` mechanism overrules the RobotFramework_TestsuitesManagement mechanisms `config_file` and `local_config`
(like `--variable <single variable>:...` does also).

For me this behavior is plausible, but I would like to have the confirmation from you, that this is what you want to have.

In case of this behavior is confirmed by you, the following tasks have to be done:

* Extend the log

It is not mentioned anywhere in the log file that another input channel for parameters (`--variablefile`) is involved:

```
Parameters loaded from 'C:\RobotTest\projects\ROBFW_AIO\robotframework-tutorial\900_building_testsuites\exercise-07\config\exercise-07_config_default.json'
CfgFile Path: C:\RobotTest\projects\ROBFW_AIO\robotframework-tutorial\900_building_testsuites\exercise-07\config\exercise-07_config_default.json
Local config file: C:\RobotTest\projects\ROBFW_AIO\robotframework-tutorial\900_building_testsuites\exercise-07\localconfig\exercise-07_localconfig_bench1.json
```

Only by taking a look at the log file a user has no chance to get to know why parameter values are not like defined within the listed configuration files.

Maybe an extension is possible in the log files like:
`Robot Framework parameter file: C:\RobotTest\projects\ROBFW_AIO\robotframework-tutorial\900_building_testsuites\exercise-07\variables\testvariables.py`

* This has to be documented in main documentation of the component and also in tutorial.

**Comment by namsonx** [comment]
Hello Holger,

I also agree with the behavior you mentioned in your comment above, it was also requested from Thomas via issue #49 
For log out the information in case robot run with variablefile, I will implement on next release.

Thank you,
Son

**Comment by namsonx** [comment]
Hello @test-fullautomation and @HolQue ,

I had checked the --variablefile is set in SuiteRunner._settings of Robot FW core which our testsuitemanagement could not access.
Do we really need to modify the core of Robot FW to get variablefile value to log it out?

Thank you,
Son

**Comment by HolQue** [comment]
Hello Son,

I agree that it is not a good idea to modify the code code because of such kind of CRQs. Maybe a fallback solution can be like this:

In documentation we explain that `config_file` and `local_config` are RobotFramework_TestsuitesManagement specific settings whereas `variablefile` is a Robot Framework setting that overrules the settings from RobotFramework_TestsuitesManagement.

Therefore we recommend not to mixup both ways to define parameter values - because the origin of parameter values is not traceable in log file any more.

**Comment by namsonx** [comment]
Hello Holger,

I agree that we just explain in our document is enough. :) 

Thank you,
Son

**Comment by test-fullautomation** [comment]
Hi Holger,

please inform me once documented that I can close the issue.

Thank you,
 Thomas

**Comment by HolQue** [comment]
Done with

https://github.com/test-fullautomation/robotframework-testsuitesmanagement/pull/115

**Comment by test-fullautomation** [comment]
solved and already merged with #115
